packages/material-ui-styles/src/createGenerateClassName/createGenerateClassName.js

```diff
@@ -39,13 +39,19 @@ export default function createGenerateClassName(options = {}) {
       [
         'Material-UI: you might have a memory leak.',
         'The ruleCounter is not supposed to grow that much.',
-      ].join(''),
+      ].join('\n'),
     );
 
     const name = styleSheet.options.name;
 
     // Is a global static MUI style?
-    if (name && name.indexOf('Mui') === 0 && !styleSheet.options.link && !disableGlobal) {
+    if (
+      name &&
+      name.indexOf('Mui') === 0 &&
+      !styleSheet.options.muiDynamic &&
```

Comment:
This was important to use different class name for every Box instance (otherwise class name of all elements will be "MuiBox-root").